06_neighborhoods/Transportation/transit_trips_cost_county_subgroups.R

```diff
@@ -0,0 +1,729 @@
+
+cost_all_15 <- cost_data_2015 %>%
+  group_by(state, county) %>%
+  summarize(index_transportation_cost = weighted.mean(x = t_80ami, w = households, na.rm = TRUE)) %>%
+  mutate(subgroup_type = "race-ethnicity",
+         subgroup = "All")
+
+cost_all_19 <- cost_data_2019 %>%
+  group_by(state, county) %>%
+  summarize(index_transportation_cost = weighted.mean(x = t_80ami, w = households, na.rm = TRUE)) %>%
+  mutate(subgroup_type = "race-ethnicity",
+         subgroup = "All")
+
+trips_all_15 <- trips_data_2015 %>%
+  group_by(state, county) %>%
+  summarize(index_transit_trips = weighted.mean(x = transit_trips_80ami, w = households, na.rm = TRUE)) %>%
+  mutate(subgroup_type = "race-ethnicity",
+         subgroup = "All")
+
+trips_all_19 <- trips_data_2019 %>%
+  group_by(state, county) %>%
+  summarize(index_transit_trips = weighted.mean(x = transit_trips_80ami, w = households, na.rm = TRUE)) %>%
+  mutate(subgroup_type = "race-ethnicity",
+         subgroup = "All")
+
+
+# Now make sure to change the trips to national percentile ranks
+
+# Create a new column 'rank' and initialize it with NA
+trips_all_15$rank <- NA
+# Calculate ranks for non-NA values
+non_na_rows <- !is.na(trips_all_15$index_transit_trips)
+trips_all_15$rank[non_na_rows] <- rank(trips_all_15$index_transit_trips[non_na_rows])
+
+# Create a new column 'rank' and initialize it with NA
+trips_all_19$rank <- NA
+# Calculate ranks for non-NA values
+non_na_rows <- !is.na(trips_all_19$index_transit_trips)
+trips_all_19$rank[non_na_rows] <- rank(trips_all_19$index_transit_trips[non_na_rows])
+
+
+# Calculate percentile ranks
+trips_all_15$percentile_rank <- (trips_all_15$rank - 1) / (nrow(trips_all_15) - 1) * 100
+trips_all_19$percentile_rank <- (trips_all_19$rank - 1) / (nrow(trips_all_19) - 1) * 100
+# round to 2 decimals
+trips_all_15$index_transit_trips <- round(trips_all_15$percentile_rank, 2)
+trips_all_19$index_transit_trips <- round(trips_all_19$percentile_rank, 2)
+# keep only what we need
+trips_all_15 <- trips_all_15 %>%
+  select(state, county, index_transit_trips, subgroup_type, subgroup)
+trips_all_19 <- trips_all_19 %>%
+  select(state, county, index_transit_trips, subgroup_type, subgroup)
+
+
+# 3b. Collapse accordingly -- to counties and race categories, weighting the measure by HH count per tract
+
+# checking if unnecessary missings are being created
+na_count <- sum(is.na(cost_data_2015$t_80ami))
+
+cost_by_race_15 <- cost_data_2015 %>%
+  group_by(state, county, race_category) %>%
+  summarize(transit_cost = weighted.mean(x = t_80ami, w = households, na.rm = TRUE))
+
+na_count <- sum(is.na(cost_by_race_15$transit_cost))
+# went from 810 missings to 196 and the values look consistent
+
+cost_by_race_19 <- cost_data_2019 %>%
+  group_by(state, county, race_category) %>%
+  summarize(transit_cost = weighted.mean(x = t_80ami, w = households, na.rm = TRUE))
+
+trips_by_race_15 <- trips_data_2015 %>%
```

Comment:
This is county-level weighted mean. Is that correct or should it be weighted sum?

Reply:
For this one, t_80ami is actually a percentage value, not a count like the other variable in the dataset (trips), so it makes sense to mean not sum